### ISSUE TITLE
Expected behaviour of `state.Verify` when no Targets metadata is found

### ISSUE THREAD

**Issue by adityasaky** [adityasaky/gittuf#46]
First raised in https://github.com/adityasaky/gittuf/pull/37#discussion_r1193990449.

Currently, `state.Verify()` returns successfully if the top level targets role (same for delegated roles) is missing after verifying the root envelope's signatures. This is designed to support initial states where a root role may exist but no policies yet. We need to revisit how to handle this check going forward.

It's an issue if an attacker is able to modify the policy state of some copy of the repository to remove targets metadata. But for this, they need to be authorized to sign the RSL entry for the policy namespace. Also, future namespace specific verification workflows will error out if no policy metadata is found so this check may be redundant (not that redundancy is bad).

**Comment by adityasaky** [comment]
> But for this, they need to be authorized to sign the RSL entry for the policy namespace.

This is not true anymore. The validity of a policy state is strictly verified internally except for its root keys. A policy state's root keys must be trusted by the previous policy state. So we need to think about how to handle cases where targets metadata are removed. We might have to special case this a bit, handling the top level role differently from the rest, and also double check how TUF behaves. AIUI, the snapshot role allows for removing a targets role if there's a key rotation to go with it. @JustinCappos can you confirm?

**Comment by JustinCappos** [comment]
In general, the snapshot doesn't allow the removal of a targets role / metadata file, but does control which versions are available and trusted.  (This is similar, but not identical to what you say above.)

I think we need to consider how to appropriately merge timestamp and snapshot into gittuf.  I believe it may not be necessary to have either, but would like to discuss.

**Comment by adityasaky** [comment]
The RSL state can be used to infer snapshot properties (minus, possibly, the signer making the snapshot claim). It could be as simple as setting some constraints on what targets roles a state must contain based on the previous one. This is akin to #116's changes but expanded to targets roles. If I understand correctly, in the default case a new policy state must contain all the same targets roles (identified by name) as the previous one, may introduce one or more new targets roles (identified by name), and all roles must have the same version or version + 1 as the prior state.

**Comment by adityasaky** [comment]
That does, however, require us to work out how targets can be removed when the delegated role is no longer necessary. This is one of a few examples we've encountered so far where gittuf's expectations deviate from TUF's. We're never verifying historic artifacts / states using the current set of metadata in gittuf, which is largely not the case with TUF.

**Comment by adityasaky** [comment]
@JustinCappos @patzielinski it'd be good to take a call on this one.
